aws/data_source_aws_api_gateway_resource.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsApiGatewayResource() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsApiGatewayResourceRead,
+		Schema: map[string]*schema.Schema{
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"path_part": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsApiGatewayResourceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	restApiId := d.Get("rest_api_id").(string)
+	target := d.Get("path").(string)
+	params := &apigateway.GetResourcesInput{RestApiId: aws.String(restApiId)}
+
+	var matches []*apigateway.Resource
```

Comment:
Is it possible to have multiple resources at the same path? If not, we should change this to be singular and `return false` in the `GetResourcesPages` after it matches to save API calls.

Reply:
Good idea.
Changing now.